File: backend/src/main/java/com/davnig/invoicemanagementapi/model/dto/TaxableSubjectDetail.java
package com.davnig.invoicemanagementapi.model.dto;

import com.davnig.invoicemanagementapi.model.entity.TaxableSubject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TaxableSubjectDetail {

    private String piva;
    private String firstName;
    private String lastName;
    private String taxCode;

    public TaxableSubjectDetail(TaxableSubject entity) {
        this.piva = entity.getPiva();
        this.firstName = entity.getFirstName();
        this.lastName = entity.getLastName();
        this.taxCode = entity.getTaxCode();
    }

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/dto/TaxableSubjectDetail.kt
package com.davnig.invoicemanagementapi.model.dto

import com.davnig.invoicemanagementapi.model.entity.TaxableSubject

class TaxableSubjectDetail {

    var piva: String? = null
    var firstName: String? = null
    var lastName: String? = null
    var taxCode: String? = null

    constructor(entity: TaxableSubject) {
        piva = entity.piva
        firstName = entity.firstName
        lastName = entity.lastName
        taxCode = entity.taxCode
    }

}